esp-wifi/src/ble/os_adapter_esp32.rs

```diff
@@ -301,11 +302,9 @@ pub(crate) fn btdm_controller_mem_init() {
             len,
         );
 
-        log::debug!(
-            "btdm_controller_mem_init {:p} {:p} {}",
-            _data_start_btdm_rom as *const u8,
-            &_data_start_btdm as *const _ as *mut u8,
-            len,
+        debug!(
```

Comment:
I'll be honest, I have no idea what is going on in this function. Why are some of the statics treated as addresses (which seems logical given their definitions), but the code takes the address of others? They are defined the same way:

```
PROVIDE ( _data_start_btdm_rom = 0x4000d4f4 );
PROVIDE ( _data_start_btdm = 0x3ffae6e0);
PROVIDE ( _data_end_btdm = 0x3ffaff10);
```

Are these turned into actual variables with actual addresses? And in that case, are we interested in those addresses instead of what is defined in the linker script? Or are these (the address and the value) the same?

Reply:
@bjoernQ do you happen to remember what's why?

Reply:
Ok - that's indeed weird. I first thought that `_data_start_btdm` and `_data_end_btdm` are defined in one of the libs and the PROVIDE will get overwritten but doesn't seem like that. Without the PROVIDE the linker will complain.

However no other way of treating the statics in a way that compiles and gets the correct results 🤷 (i.e. no way to treat them the same)

I thought the linker does funky things but in the linked elf file the symbols look same:
```
❯ nm ..\target\xtensa-esp32-none-elf\debug\examples\ble ^_data_start_btdm
3ffae6e0 A _data_start_btdm
4000d4f4 A _data_start_btdm_rom
```

The big difference is that `_data_start_btdm_rom` is just a random address for the linker (we don't define ROM as memory) while `_data_start_btdm` points to "valid" memory. Maybe that's what is causing the "magic"?

In the end in all cases, we need the address

Reply:
Thanks! It's weird, but I think I can see the logic of hiding rom data behind a pointer. I've created a PR that touches up the surroundings of this bit, hopefully for the better.

Reply:
Yes! It's mainly to avoid maintaining different addresses for different revisions. Same with a most ROM functions which go through an indirection with fixed addresses